[CS2113-T18-4] NUScents

lckjosh opened the pull request:
Welcome to NUScents, the tailor-made financial tracker for SOC students at NUS. It is optimized for use via a Command Line Interface (CLI) to offer a clutter-free solution for our users to manage and monitor their financial activities.

irving11119 commented on src/main/java/seedu/nuscents/parser/Parser.java:
```diff
@@ -0,0 +1,300 @@
+            AllowanceCategory allowanceCategory = parseAllowanceCategory(category);
+            String format = datePatternValidation(date);
+            SimpleDateFormat formatter = new SimpleDateFormat(format);
+            Date formattedDate = parseDate(date, format, formatter);
```
Perhaps you could abstract this out into a separate method, especially seeing how its quite similar to the code below.

irving11119 commented on src/main/java/seedu/nuscents/storage/Storage.java:
```diff
@@ -0,0 +1,165 @@
+                break;
+            }
+        }
+    }
```
This method is a bit length, consider if you can use more abstraction to make it cleaner and shorter!

irving11119 commented on src/main/java/seedu/nuscents/Nuscents.java:
```diff
@@ -0,0 +1,75 @@
+                ui.showException(e);
+            } catch (IOException | ParseException e) {
+                Ui.showLine();
+                System.out.println("Something went wrong: " + e.getMessage());
```
Perhaps printing of error message here can also go into your UI class for good SLAP.

irving11119 commented on src/main/java/seedu/nuscents/data/TransactionList.java:
```diff
@@ -0,0 +1,91 @@
+
+        if (isFound) {
+            Ui.showFilterMessage(filteredTransactions, category);
+            System.out.println("Filtered transactions in the category " + category + ":");
```
Perhaps your system output here can be moved to the UI class? This ensures each class has follows Single Responsibility Principle e.g. printing to the console is handled solely by the UI class.

irving11119 commented on src/main/java/seedu/nuscents/ui/Messages.java:
```diff
@@ -0,0 +1,43 @@
+package seedu.nuscents.ui;
+
+public class Messages {
```
Good use of a class to store your constant messages to be printed

irving11119 reviewed (commented):
Overall your code has decent abstraction and follow good OOP principles. However, continue looking into how you can refactor your code to make it more readable and neater!

spaceman03 committed:
Merge pull request #59 from spaceman03/user-guide-v2.0

Update UserGuide.md with quick start and command summary

aaronxujiachen committed:
Update User Guide with all current features.

aaronxujiachen committed:
Update User Guide with all current features.

aaronxujiachen committed:
Delete src/main/java/seedu/nuscents/commands/EditCommand.java

aaronxujiachen committed:
Merge pull request #61 from aaronxujiachen/branch-editCommand

Update User Guide with all current features.

lckjosh committed:
Add hmac verification for storage file

lckjosh committed:
Fix IO redirection test

spaceman03 committed:
Update sequence diagram for add transaction

spaceman03 committed:
Update DeveloperGuide.md with Architecture Diagram

spaceman03 committed:
Merge pull request #65 from spaceman03/developer-guide

Update developer guide with architecture diagram and minor fixes

lckjosh committed:
Merge pull request #64 from lckjosh/add-hmac-storage-file

Add hmac verification for storage file

yingx9 commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,180 @@
-|--------|----------|---------------|------------------|
-|v1.0|new user|see usage instructions|refer to them when I forget how to use the application|
-|v2.0|user|find a to-do item by name|locate a to-do without having to go through the entire list|
+| Version | As a ...           | I want to ...                                             | So that I can ...                                        |
+|---------|--------------------|-----------------------------------------------------------|----------------------------------------------------------|
```
<img width="791" alt="image" src="https://github.com/nus-cs2113-AY2324S1/tp/assets/84222810/eadc8348-b491-4c20-a0aa-e9b76d6e6ef7">

I think your table formatting is broken in the github.io site

yingx9 commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,180 @@
+
+The following sequence diagram shows how the add transaction operation works:
+
+<img src="images/AddTransactionSequenceDiagram.png" width="600" />
```
<img width="514" alt="image" src="https://github.com/nus-cs2113-AY2324S1/tp/assets/84222810/31999f8b-766d-4873-ae4e-a20112375ab2">

Maybe you can omit the classes and User at the bottom because it might not be necessary to repeat them again? Same for the other sequence diagrams

DavinciDelta replied:
<img width="467" alt="image" src="https://github.com/nus-cs2113-AY2324S1/tp/assets/110587475/ecca6d9e-0620-4b90-bf08-0084ee19d931">

I think it is bit hard to read the words in the red boxes...

DavinciDelta replied:
Is the classes at the bottom required? I think it makes it confusing

DavinciDelta replied:
so when a parseCommand is called, a new addCommand and expense class is created? Should it not have the constructor method?

lckjosh committed:
Update UG to include categories for filter command and storage file FAQs

lckjosh committed:
Merge pull request #66 from lckjosh/update-ug

Update UG to include categories for filter command and storage file FAQs

yingx9 commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,180 @@
+However, we opted for the current design to promote a cleaner separation of concerns and to facilitate future expansions 
+and modifications.
+
+The following sequence diagram shows how the add transaction operation works:
+<img src="images/ViewSequenceDiagram.png" width="600" />
```
The sentence says "how the add transaction operation works" but the sequence diagram is about the ViewCommand. Maybe that's an error?

<img width="496" alt="image" src="https://github.com/nus-cs2113-AY2324S1/tp/assets/84222810/9a7f2330-dd60-4b1e-8937-b0e1df39f133">

DavinciDelta replied:
<img width="467" alt="image" src="https://github.com/nus-cs2113-AY2324S1/tp/assets/110587475/4322d33f-e326-4840-9a72-6ad3a610696e">

Is a new ViewCommand created each time parseCommand is called?

yingx9 commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,180 @@
+6. UI: Handles user interface and messaging.
+
+#### II. Component-Level Design
+1. Parser
+
+The Parser class identifies the "view" command and extracts the taskIndex (transaction index) from the user's input.
+
+2. ViewCommand
+
+The ViewCommand object is created by the Parser. It encapsulates the user's request to view a specific transaction. This object is passed to the Nuscents class for execution.
+
+3. Nuscents
+
+In the Nuscents class, the execute() method of the ViewCommand is called, and the taskIndex is extracted from the command. It then calls the viewTransaction(taskIndex) method on the TransactionList.
```
![image](https://github.com/nus-cs2113-AY2324S1/tp/assets/84222810/2ae0d837-1595-47bc-9705-87407a091498)

The numbering is broken in the github.io site

yingx9 reviewed (commented):
Overall a nice DG with great and understandable sequence diagrams, only minor formatting mistakes

spaceman03 committed:
Fix user guide

spaceman03 committed:
Merge pull request #67 from spaceman03/user-guide-v2.0

Fix user guide

spaceman03 committed:
Fix numbering typo in UserGuide.md

spaceman03 committed:
Merge pull request #68 from spaceman03/user-guide-v2.0

Fix numbering typo in UserGuide.md

spaceman03 committed:
Fix table format

spaceman03 committed:
Merge pull request #69 from spaceman03/developer-guide

Fix table format

spaceman03 committed:
Fix table format to ensure correct format in deployment

spaceman03 committed:
Merge pull request #70 from spaceman03/developer-guide

Fix table format to ensure correct format in deployment

spaceman03 committed:
Fix spaces between columns

spaceman03 committed:
Merge pull request #71 from spaceman03/developer-guide

Fix spaces between columns

BenedictChannn commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,180 @@
+
+Given below is the example usage scenario and how the list transaction mechanism behaves at each step.
+
+Step 1. The user launches the application. The `TransactionList` will be initialized with the transactions stored in 
```
Would it be good to extract this file storage and retrieval out to a section on its own to explain how it works?

BenedictChannn commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,180 @@
+
+### `helpCommand` Usage Scenario
+
+1. **Step 1**: The user launches the application. The initial screen appears.
```
Not sure if "1. Step 1" is necessary. would using "Step 1" suffice?

BenedictChannn commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,180 @@
-|--------|----------|---------------|------------------|
-|v1.0|new user|see usage instructions|refer to them when I forget how to use the application|
-|v2.0|user|find a to-do item by name|locate a to-do without having to go through the entire list|
+Version | As a ... | I want to ...                                             | So that I can ...
```
The formatting for the User Stories seems to be broken on the page

lckjosh committed:
Fix table borders in UG

lckjosh committed:
Merge pull request #72 from lckjosh/fix-ug-table

Fix table borders in UG

lckjosh committed:
Hide footbox for UML sequence diagrams

lckjosh committed:
Merge pull request #73 from lckjosh/master

Hide footbox for UML sequence diagrams

aaronxujiachen committed:
Add EditCommand Feature

aaronxujiachen committed:
Update EditCommand.java

choonkit-nus committed:
Update net balance feature to follow SRP

choonkit-nus committed:
Merge pull request #75 from choonkit-nus/branch-netbalance-update

Update net balance feature to follow SRP

choonkit-nus committed:
Add budget feature

choonkit-nus committed:
Fix gradle build check for excessive imports

choonkit-nus committed:
Update budget feature to reflect whether expenses are within or exceeded

choonkit-nus committed:
Update help message to reflect budget feature

choonkit-nus committed:
Update EXPECTED.TXT for text-ui-test

aaronxujiachen committed:
Update EditCommand.java

aaronxujiachen committed:
Update TransactionList.java

choonkit-nus committed:
Add persistent storage for budget feature

vvhuiling committed:
Update view feature to show category

choonkit-nus committed:
Fix coding standard to improve readability

vvhuiling committed:
Add JUnit for TransactionList

vvhuiling committed:
Merge pull request #78 from vvhuiling/Update-view-feature

Update view feature to show category

vvhuiling committed:
add test case to TransactionListTest

vvhuiling committed:
Bug fix: ignore space and empty line

aaronxujiachen committed:
Merge pull request #74 from aaronxujiachen/branch-editCommand

Add EditCommand Feature

choonkit-nus committed:
Merge branch 'master' into branch-budget

choonkit-nus committed:
Update Parser.java

choonkit-nus committed:
Merge pull request #77 from choonkit-nus/branch-budget

Add budget feature

vvhuiling committed:
Bug fix: standardize line ending

vvhuiling committed:
Bug fix: standardize line ending

vvhuiling committed:
Bug fix: standardize line ending

aaronxujiachen committed:
Update Help Menu and User Guide for Edit Command

vvhuiling committed:
Bug fix: standardize line ending

aaronxujiachen committed:
Merge pull request #82 from aaronxujiachen/branch-helpCommand

Update Help Menu and User Guide for Edit Command

vvhuiling committed:
update the expectedOutput

vvhuiling committed:
Merge pull request #81 from vvhuiling/JUnit-TransactionListTes

Junit transactionList test

DaDevChia commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,181 @@
+Step 1. The user launches the application for the first time. The `TransactionList` will be initialized.
+
+Step 2. The user executes `expense /amt 20 /date 24-10-2023 /desc Lunch /note Pasta /cat Food` command to create a
+transaction. The `expense` command calls `Parser#parseExpense()` to create an `Expense` object. The 
+`AddCommand#execute()` is then called to store the `Expense` object in the `TransactionList`.
```
Should this be Parser.parseExpense() and AddCommand.execute()?

Barbaracwx commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,181 @@
 
 ## Acknowledgements
-
 {list here sources of all reused/adapted ideas, code, documentation, and third-party libraries -- include links to the original source as well}
```
Maybe before explaining the design and implementation you can add in a brief summary of each component's functions and responsibilities for easier understanding of the system

DaDevChia commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,181 @@
+
+6. UI
+
+The UI class displays the transaction details using the UI.showTransactionViewMessage method, which receives the Transaction object as input and presents the detailed transaction information to the user.
```
For greater clarity should UI.showTransactionViewMessage() instead of UI.showTransactionViewMessage be used to indicate that it is a method more clearly?

DaDevChia commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,181 @@
+
+The following sequence diagram shows how the list transaction operation works:
+
+<img src="images/ListTransactionSequenceDiagram.png" width="600" />
```
![Uploading Screenshot 2023 -11-02 at 5.39.34 PM.jpg…]()
Are the duplicate classes at the bottom of the diagram required? Could make it confusing in terms of sequence flow.

DaDevChia reviewed (commented):
Looks great, just a few minor formatting errors and points for improvements in sequence diagram.

spaceman03 committed:
Update UserGuide.md to include add budget feature

spaceman03 committed:
Merge pull request #83 from spaceman03/user-guide-v2.0

Update UserGuide.md to include add budget feature

StanleyW00 commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,181 @@
+
+The following sequence diagram shows how the add transaction operation works:
+
+<img src="images/AddTransactionSequenceDiagram.png" width="600" />
```
![image](https://github.com/nus-cs2113-AY2324S1/tp/assets/110610562/8579f6b1-0018-492e-99da-c1d121b3809a)
I think that you can omit the user and the boxes below the sequence diagram as it is unnecessary. The same goes for the other sequence diagrams below.

StanleyW00 replied:
![image](https://github.com/nus-cs2113-AY2324S1/tp/assets/110610562/125731ae-3c03-42b6-b1c2-6c09a1b6329b)
For the arrow with `parseExpense("/amt...")`, I believe that you should use a dashed arrow line for this part.

StanleyW00 commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,181 @@
+
+### `helpCommand` Usage Scenario
+
+1. **Step 1**: The user launches the application. The initial screen appears.
+2. **Step 2**: Unsure of the commands, the user inputs the `help` command.
+3. **Step 3**: The application recognizes the command through the `Parser` and creates a `HelpCommand` object.
+4. **Step 4**: The `Nuscents` class invokes the `execute()` method of the `HelpCommand`.
+5. **Step 5**: The `UI` fetches the `HELP_MENU` string and displays the comprehensive list of commands to the user.
```
I think that you can omit either `1.` or `Step 1` as it is redundant to use both. This also applies for the rest of the points in list of usage scenarios.

StanleyW00 commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,181 @@
+In the Nuscents class, the execute() method of the ViewCommand is called, and the taskIndex is extracted from the command. It then calls the viewTransaction(taskIndex) method on the TransactionList.
+
+4. TransactionList
+
+The TransactionList contains a list of Transaction objects. The viewTransaction(taskIndex) method retrieves the specific Transaction object based on the taskIndex.
+
+5. Transaction
 
-{Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
+The Transaction class represents an individual transaction, and it contains all the relevant details of a transaction.
+
+6. UI
```
![image](https://github.com/nus-cs2113-AY2324S1/tp/assets/110610562/344b2a53-b369-40b9-9053-a84ff684e571)
There is a bug where the numbering for the list is only in `1`s.

StanleyW00 reviewed (commented):
LGTM. Just a few nits needed to fix. Perhaps you should add explanations for the storage as well.

spaceman03 committed:
Update UserGuide.md to provide more information for the budget feature

spaceman03 committed:
Update budget invalid message

spaceman03 committed:
Merge pull request #128 from spaceman03/user-guide-v2.0

Update UserGuide.md to provide more information for the budget feature

spaceman03 committed:
Bug fix: Standardize formatting

spaceman03 committed:
Merge pull request #129 from spaceman03/user-guide-v2.0

Bug fix: Standardize formatting

spaceman03 committed:
Remove duplication of class names in sequence diagram

spaceman03 committed:
Bug fix: numbering issue

spaceman03 committed:
Merge pull request #134 from spaceman03/developer-guide

Update DeveloperGuide.md to fix numbering and typo issues

spaceman03 committed:
Update numbering of DeveloperGuide.md

spaceman03 committed:
Merge pull request #135 from spaceman03/developer-guide

Update numbering of DeveloperGuide.md

spaceman03 committed:
Update UserGuide.md with table of contents

spaceman03 committed:
Update command summary

spaceman03 committed:
Add visuals to UserGuide.md

spaceman03 committed:
Update README.md

spaceman03 committed:
Merge pull request #136 from spaceman03/user-guide-v2.0

Update UG with table of content and command summary

spaceman03 committed:
Draft of PPP for spaceman03

spaceman03 committed:
Merge pull request #137 from spaceman03/project-portfolio-page

Draft of PPP for spaceman03

spaceman03 committed:
Bug fix: Make transaction details align with header

spaceman03 committed:
Bug fix: Add a space for displaying allowance

spaceman03 committed:
Merge pull request #138 from spaceman03/list-feature

Bug fix: Fix alignment issue when a long description is given

spaceman03 committed:
Fix formatting issue for UserGuide.md

spaceman03 committed:
Add separator between Table of Contents and Introduction

spaceman03 committed:
Add note to remind user

spaceman03 committed:
Add screenshots to UserGuide.md

spaceman03 committed:
Merge pull request #139 from spaceman03/user-guide-v2.0

Update UserGuide.md with screenshots and fix formatting issues

spaceman03 committed:
Move count method from Transaction to TransactionList

spaceman03 committed:
Remove redundant comment

spaceman03 committed:
Add JUnit test for AddCommand class

spaceman03 committed:
Add JUnit test for DeleteCommand class

spaceman03 committed:
Remove unused import statement

spaceman03 committed:
Add JUnit test for EditCommand class

spaceman03 committed:
Edit date in transactions in command testing

spaceman03 committed:
Rename parameter from tasks to transactions

spaceman03 committed:
Remove negative symbol in expected output

spaceman03 committed:
Edit filter category display message

spaceman03 committed:
Add JUnit test for FilterCommand class

spaceman03 committed:
Remove unused import statements

spaceman03 committed:
Reorder variables

spaceman03 committed:
Edit expected output

spaceman03 committed:
Add setup before each test in ParserTest

spaceman03 committed:
Replace star import with explicit import statements

spaceman03 committed:
Add JUnit test for BudgetCommand class

spaceman03 committed:
Add JUnit test for HelpCommand class

spaceman03 committed:
Add JUnit test for InvalidCommand class

spaceman03 committed:
Add JUnit test for ListCommand class

spaceman03 committed:
Add JUnit test for ViewCommand class

spaceman03 committed:
Remove unused import statements

spaceman03 committed:
Move code to next line to abide java coding standards

spaceman03 committed:
Merge pull request #140 from spaceman03/command-test

Add JUnit test for command classes

choonkit-nus committed:
Update budget feature to only accept float that doesn't exceed 2 d.p.

choonkit-nus committed:
Update parseAllowance, parseExpense, parseBudget methods

Added a check to the parseAllowance(), parseExpense() and parseBudget() methods in Parser.java to ensure that the following conditions are met for the user entered amount. The allowance, expense and budget amount is positive, less than 100000, and has at most two decimal places. This ensures that arbitrary large or small float amounts are rejected and not used.

choonkit-nus committed:
Update budget feature to execute showBudgetSet to only when a new budget is set

choonkit-nus committed:
Fix bug related to incorrect transactionList count

choonkit-nus committed:
Remove excess new line between method

choonkit-nus committed:
Merge pull request #141 from choonkit-nus/branch-budget-update

Branch budget update

lckjosh committed:
create lckjosh PPP

lckjosh committed:
Update lckjosh PPP

lckjosh committed:
Merge pull request #142 from lckjosh/master

create lckjosh PPP

lckjosh committed:
Remove hmac verification to follow tp constraints

lckjosh committed:
Merge pull request #144 from lckjosh/remove-hmac

Remove hmac verification to follow tp constraints

lckjosh committed:
Fix exception if no permission to access data file

lckjosh committed:
Fix line length

lckjosh committed:
Merge pull request #145 from lckjosh/remove-hmac

Fix exception if no permission to access data file

lckjosh committed:
Handle corrupted budget file

lckjosh committed:
Merge pull request #146 from lckjosh/master

Handle corrupted budget file

vvhuiling committed:
Update user guide

vvhuiling committed:
Merge pull request #148 from vvhuiling/DG

Update developer guide

vvhuiling committed:
Update user guide

vvhuiling committed:
Merge pull request #149 from vvhuiling/UG

Update user guide

lckjosh committed:
Add the ability to handle corrupted data files

lckjosh committed:
Merge pull request #151 from lckjosh/handle-corrupted-data

Add the ability to handle corrupted data files

vvhuiling committed:
BugFix: specify OTHERS

vvhuiling committed:
Merge pull request #152 from vvhuiling/Bugfix

BugFix: specify OTHERS

vvhuiling committed:
create vvhuiling PPP

vvhuiling committed:
Merge pull request #153 from vvhuiling/PPP

create vvhuiling PPP

lckjosh committed:
Remove code duplication in transactionDecoder()

lckjosh committed:
Fix line length

lckjosh committed:
Merge pull request #154 from lckjosh/master

Remove code duplication in transactionDecoder()

lckjosh committed:
Update count to consider transactions loaded from data file

lckjosh committed:
Merge pull request #155 from lckjosh/master

Update count to consider transactions loaded from data file

spaceman03 committed:
Update PPP for spaceman03

spaceman03 committed:
Upload picture for Zhi Xuan

spaceman03 committed:
Merge pull request #156 from spaceman03/project-portfolio-page

Update PPP for spaceman03

spaceman03 committed:
Bug fix: show invalid date message when user inputs invalid date

spaceman03 committed:
Remove unused parameter

spaceman03 committed:
Remove unused method

aaronxujiachen committed:
Update Error Message For Edit Command

aaronxujiachen committed:
Update ParserTest For EditCommand

aaronxujiachen committed:
Merge pull request #159 from aaronxujiachen/branch-editCommand

Update Error Message And ParserTest For EditCommand

spaceman03 committed:
Bug fix

spaceman03 committed:
Merge branch 'master' into bug-fix

spaceman03 committed:
Bug fix: date validation

spaceman03 committed:
Merge branch 'bug-fix' of https://github.com/spaceman03/tp into bug-fix

spaceman03 committed:
Merge pull request #157 from spaceman03/bug-fix

Bug fix: shows invalid date message when inputting invalid date

spaceman03 committed:
Update AboutUs.md for spaceman03

spaceman03 committed:
Merge pull request #161 from spaceman03/project-portfolio-page

Update AboutUs.md for spaceman03

spaceman03 committed:
Update README.md

spaceman03 committed:
Merge pull request #162 from spaceman03/user-guide-v2.0

Update README.md

spaceman03 committed:
Update Architecture section in DG

choonkit-nus committed:
Update DG with NFR

choonkit-nus committed:
Update DG with App introduction

choonkit-nus committed:
Update DG with Design section core components and interaction flow

choonkit-nus committed:
Update DG with value proposition

choonkit-nus committed:
Update DG with rename Model component to Data

spaceman03 committed:
Add UI and Data component to the DG

choonkit-nus committed:
Update DG with budget feature

choonkit-nus committed:
Merge pull request #163 from choonkit-nus/branch-developer-guide-NFR

Branch developer guide NFR

spaceman03 committed:
Bug fix: enable user input to be case insensitive

spaceman03 committed:
Merge branch 'master' into developer-guide

spaceman03 committed:
Edit add transaction sequence diagram

spaceman03 committed:
Merge branch 'developer-guide' of https://github.com/spaceman03/tp into developer-guide

spaceman03 committed:
Export new add transaction sequence diagram

spaceman03 committed:
Merge pull request #164 from spaceman03/developer-guide

Update DG design section

spaceman03 committed:
Bug fix: enable case insensitive arguments

spaceman03 committed:
Merge pull request #165 from spaceman03/bug-fix

Bug fix: enable user input to be case insensitive

spaceman03 committed:
Bug fix

spaceman03 committed:
Merge pull request #166 from spaceman03/developer-guide

Bug fix for DG

lckjosh committed:
Add class diagram for Command component

lckjosh committed:
Include <<interface>> and <<enumeration>> for Data class diagram

lckjosh committed:
Merge pull request #167 from lckjosh/master

Add Command class diagram

lckjosh committed:
Update DataClassDiagram.png

lckjosh committed:
Merge pull request #168 from lckjosh/master

Update DataClassDiagram.png

aaronxujiachen committed:
Update JUnit Test And Developer Guide For Edit Command

aaronxujiachen committed:
Merge pull request #169 from aaronxujiachen/branch-editCommand

Update JUnit Test And Developer Guide For Edit Command

lckjosh committed:
Update Interaction Flow

aaronxujiachen committed:
PPP FOR XU JIACHEN

aaronxujiachen committed:
Merge pull request #171 from aaronxujiachen/branch-editCommand

PPP FOR XU JIACHEN

aaronxujiachen committed:
Update aaronxujiachen.md

aaronxujiachen committed:
Update aaronxujiachen.md

lckjosh committed:
Merge pull request #170 from lckjosh/master

Update Interaction Flow

choonkit-nus committed:
Add choonkit-nus PPP

lckjosh committed:
Link PPP for lckjosh in AboutUs

choonkit-nus committed:
Update AboutUs.md

lckjosh committed:
Add profile photo for AboutUs for lckjosh

spaceman03 committed:
Update DeveloperGuide.md

lckjosh committed:
Update sequence diagrams and AboutUs

spaceman03 committed:
Fix formatting issues

spaceman03 committed:
Merge pull request #173 from spaceman03/developer-guide

Update DeveloperGuide.md

lckjosh committed:
Merge pull request #172 from lckjosh/master

Update DG

choonkit-nus committed:
Merge branch 'master' into branch-choonkit-nus-PPP

choonkit-nus committed:
Update AboutUs.md

choonkit-nus committed:
Merge pull request #174 from choonkit-nus/branch-choonkit-nus-PPP

Branch choonkit-nus PPP

spaceman03 committed:
Bug fix: ensure consistency of user input

spaceman03 committed:
Bug fix: ensure invalid date will prompt an error message

vvhuiling committed:
Update DG

vvhuiling committed:
Merge pull request #176 from vvhuiling/Update-DG

Update DG

vvhuiling committed:
Update About.us

vvhuiling committed:
Merge branch 'master' into Update-about.us

vvhuiling committed:
Merge pull request #177 from vvhuiling/Update-about.us

Update About.us

vvhuiling committed:
Update PPP

vvhuiling committed:
Merge pull request #178 from vvhuiling/Update-PPP

Update PPP

spaceman03 committed:
Remove unused import statements and parameters

spaceman03 committed:
Extract printing messages to Ui class

spaceman03 committed:
Merge pull request #175 from spaceman03/bug-fix

Bug fix

spaceman03 committed:
Update note section in UG

vvhuiling committed:
Update PPP

vvhuiling committed:
Merge pull request #180 from vvhuiling/master

Update PPP

aaronxujiachen committed:
Update AboutUs.md

aaronxujiachen committed:
Merge branch 'master' into branch-editCommand

aaronxujiachen committed:
Merge pull request #181 from aaronxujiachen/branch-editCommand

Update AboutUs.md

aaronxujiachen committed:
Merge pull request #179 from spaceman03/user-guide-v2.1

Update note section in UG

aaronxujiachen committed:
Add Imgaes For AboutUs.md

aaronxujiachen committed:
Merge pull request #182 from aaronxujiachen/branch-editCommand

Add Imgaes For AboutUs.md

lckjosh committed:
Include more class diagrams

lckjosh committed:
Merge pull request #183 from lckjosh/master

Include more class diagrams

lckjosh committed:
Update Implementation: list transactions feature

lckjosh committed:
Merge pull request #184 from lckjosh/master

Update Implementation: list transactions feature

lckjosh committed:
Update UG FAQ

lckjosh committed:
Update lckjosh PPP

lckjosh committed:
Merge pull request #185 from lckjosh/master

Update UG FAQ and PPP

lckjosh committed:
Fix reposense link

lckjosh committed:
Merge pull request #186 from lckjosh/master

Fix reposense link

spaceman03 committed:
Update PPP for spaceman03

spaceman03 committed:
Merge pull request #188 from spaceman03/project-portfolio-page

Update PPP for spaceman03

spaceman03 committed:
Fix github pages formatting issue

spaceman03 committed:
Merge pull request #189 from spaceman03/developer-guide

Fix github pages formatting issue

spaceman03 committed:
Add note in UserGuide.md

spaceman03 committed:
Merge pull request #190 from spaceman03/user-guide-v2.1

Add note in UserGuide.md

lckjosh committed:
Fix format in DG list implementation steps

spaceman03 committed:
Add table of contents

spaceman03 committed:
Merge pull request #192 from spaceman03/developer-guide

Add table of contents to DG

lckjosh committed:
Add instructions for manual testing

lckjosh committed:
Merge branch 'master' into master

spaceman03 committed:
bug fix: formatting issues

lckjosh committed:
Merge pull request #191 from lckjosh/master

Fix format in DG list implementation steps, add instructions for manual testing

lckjosh committed:
Update lckjosh.md

spaceman03 committed:
Merge pull request #193 from spaceman03/project-portfolio-page

PPP bug fix: formatting issues

spaceman03 committed:
Update README.md

lckjosh committed:
Merge pull request #194 from spaceman03/user-guide-v2.1

Update README.md

spaceman03 committed:
bug fix: update links

spaceman03 committed:
Merge pull request #195 from spaceman03/user-guide-v2.1

bug fix: update links in README.md
